iis-common/src/main/java/eu/dnetlib/iis/common/java/io/SequenceFileTextValueReader.java

```diff
@@ -0,0 +1,168 @@
+import org.apache.hadoop.io.Text;
+import org.apache.hadoop.io.Writable;
+import org.apache.hadoop.util.ReflectionUtils;
+import org.apache.log4j.Logger;
+
+/**
+ * Iterator that extracts sequence file's consecutive {@link Text} values. 
+ * 
+ * @author mhorst
+ */
+public class SequenceFileTextValueReader implements CloseableIterator<Text> {
+
+        private final Logger log = Logger.getLogger(this.getClass());
+
+        private SequenceFile.Reader sequenceReader;
+
+        private final RemoteIterator<LocatedFileStatus> fileIt;
+
+        private final FileSystem fs;
+
+        /** Ignore file starting with underscore. Such files are also ignored by
+         * default by map-reduce jobs.
+         */
+        private final static String whitelistRegexp = "^(?!_).*";
+
+        private Text toBeReturned;
+
+        /**
+         * Default constructor.
+         * 
+         * @param fs hadoop filesystem
+         * @param seqPath sequence file path
+         * @throws IOException
+         */
+        public SequenceFileTextValueReader(final FileSystemPath path) 
+                        throws IOException {
+                this.fs = path.getFileSystem();
+                if (fs.isDirectory(path.getPath())) {
+                        fileIt = fs.listFiles(path.getPath(), false);
+                        sequenceReader = getNextSequenceReader();
+                } else {
+                        fileIt = null;
+                        sequenceReader = new Reader(fs.getConf(),
+                                        SequenceFile.Reader.file(path.getPath()));
+                }
+        }
+
+        final Reader getNextSequenceReader() throws IOException {
+                while (fileIt != null && fileIt.hasNext()) {
+                        LocatedFileStatus currentFileStatus = fileIt.next();
+                        if (isValidFile(currentFileStatus)) {
+                                return new Reader(this.fs.getConf(),
+                                                SequenceFile.Reader.file(currentFileStatus.getPath()));
+                        }
+                }
+//              fallback
+                return null;
+        }
+
+        /**
+         * Checks whether file is valid candidate.
+         * @param fileStatus
+         * @return true when valid, false otherwise
+         */
+        private final boolean isValidFile(LocatedFileStatus fileStatus) {
+                if (fileStatus.isFile()) {
+                        if (whitelistRegexp!=null) {
+                                return Pattern.matches(whitelistRegexp, 
+                                                fileStatus.getPath().getName());
+                        } else {
+                                return true;
+                        }
+                }
+//              fallback
+                return false;
+        }
+
+        /**
+         * Returns next data package.
+         * @return next data package
+         * @throws IOException
```

Comment:
> Doesn't throw IOException.

Fixed.

> (Aside: maybe CloseableIterator should be like/extend RemoteIterator?)

Yes, we could since `remove()` is never supported. But I would prefer to do this in separate task because this would require changes in several places where we currently expect `java.util.Iterator` instance (e.g. `eu.dnetlib.iis.wf.referenceextraction.dataset.DatasetDBBuilder` and `eu.dnetlib.iis.wf.referenceextraction.project.ProjectDBBuilder`). In this task we should also check whether it is ok to expect plain `java.util.Iterator` since it should be closed in the end...